pkg/chartmuseum/server/multitenant/index.go

```diff
@@ -40,9 +42,12 @@ func (server *MultiTenantServer) getIndexFile(log cm_logger.LoggingFn, repo stri
 	}
 
 	entry.RepoLock.Lock()
+
 	defer entry.RepoLock.Unlock()
-	// if cache is nil, and not on a timer, regenerate it
-	if len(entry.RepoIndex.Entries) == 0 && server.CacheInterval == 0 {
+	// if the keep-chart-always-update-to-date flag is set, we always update the index file
+	// and ignore the chart cache
+	if server.KeepChartAlwaysUpToDate /* the flag is set */ ||
+		(!server.KeepChartAlwaysUpToDate && len(entry.RepoIndex.Entries) == 0) /* initial */ {
```

Comment:
The only thing that worries me here is the problem i described [here](https://github.com/orgs/helm/teams/chartmuseum-maintainers/discussions/4). I think setting this flag will make the symptoms of https://github.com/helm/chartmuseum/issues/453 worse.

Reply:
Maybe we should just init the cacheEntry once , so this flag can actually ignore the cache , and skip the internal cache or external cache operation , always returns the realtime index .

Reply:
Yeah , If I understand your worry right we will have the correctness at last , say ,  for external cache like Redis , the write cache repo entry lock and the read cache repo entry lock points to the different lock , it will be not correct after upload the chart at once , but due to the new flag , we will always set the new correct repo index to the cache entry , so the cached index entry will finally be correct , right ?